### ISSUE TITLE
Terminal font ligature cannot be enabled as config

### ISSUE THREAD

**Issue by MicroBlueee** [zed-industries/zed#11127]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

the font_feature ligature cannot work properly as config in setting.

My setting json is:
```json
"terminal": {
    "alternate_scroll": "off",
    "blinking": "terminal_controlled",
    "font_family": "Zed Mono",
    "font_size": 14,
    "font_features": {
      "liga": true,
      "calt": true
    },
    "option_as_meta": true,
    "shell": {
      "program": "/usr/local/bin/fish"
    }
  },
```
 but in terminal
<img width="321" alt="截屏2024-04-28 20 17 54" src="https://github.com/zed-industries/zed/assets/49147382/8fbe61f6-c2d3-4807-b946-a7a3553b3c71">

### Environment

Zed: v0.132.3 (Zed)
OS: macOS 14.4.1
Memory: 64 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

if this setting works properly, it should be like:
<img width="216" alt="image" src="https://github.com/zed-industries/zed/assets/49147382/943f4b4a-becf-4f9d-8d9c-29290c29e196">

**Comment by liudi4046** [comment]
same

**Comment by iMonZ** [comment]
Works for me!
macOS 15 arm64

**Comment by 9999years** [comment]
I am seeing ligatures in my editor buffer but not in the terminal buffer:

<img width="260" alt="Image" src="https://github.com/user-attachments/assets/4375be98-c30f-47ff-83e2-0d9369c35984" />

This seems to happen if I don't set the `terminal` settings at all (letting Zed use the `buffer` font settings by default) or if I set the `terminal` settings identically to the `buffer` font settings explicitly.

Zed 0.174.6 d5fde6645afe6356ef31611a76aff69a3e450260
macOS Sequoia 15.3.1 (24D70)

**Comment by theherk** [comment]
Yeah, unfortunately lazygit looks like trash now. It used to look fine, now the ligatures are gone and it looks quite bad.

```
Zed: v0.183.8 (Zed Preview) 
OS: macOS 15.4.1
Memory: 32 GiB
Architecture: aarch64
```

![Image](https://github.com/user-attachments/assets/f60eaf9b-e154-4002-bf1b-677abff24398)

**Comment by FranklinYu** [comment]
I’m also affected by this issue. Ligature works in the editor but not in the integrated terminal, same as @9999years described above.

**Comment by injust** [comment]
This is still an issue in the latest stable, however this should probably be a feature request instead.

OP is a bit outdated now -- looking in the default settings, the font feature settings are `buffer_font_features` and `ui_font_features`, so it's clearer that this isn't supported in the terminal yet.

And Alacritty doesn't support ligatures AFAIK?

**Comment by mmal3k** [comment]
I was **not able to reproduce** this following the steps in the report.

### Environment
- **OS:** macOS 26.4 (25E246)
- **Hardware:** Apple M5 Pro (arm64, 24 GB RAM)
- **Zed:** 1.16.1 (stable)

### Steps taken
1. Set both the editor and terminal font-feature settings to the same values, with the same font, so the two surfaces can be compared directly:
   ```json
   "buffer_font_features": { "calt": true, "liga": true },
   "terminal": {
     "font_family": "Zed Mono",
     "font_features": { "calt": true, "liga": true }
   }
   ```
2. Put a line of ligature-forming sequences in a file and opened it in the editor:
   ```
   =>  ->  <-  <=  >=  !=  ===  !==  <=>  |>  ::  ++  --  /*  */  //  ...
   ```
3. Ran `cat` on the same file in Zed's integrated terminal, so both panes render identical bytes in the same font.
4. Flipped `buffer_font_features` to `{ "calt": false, "liga": false }`, saved, and observed both panes.
5. Flipped `terminal.font_features` to `{ "calt": false, "liga": false }`, saved, re-ran `cat`, and observed again.

### Result
Ligatures now render in the integrated terminal on 1.16.1 — so the original symptom in this report no longer occurs. However, `terminal.font_features` appears to have no effect in either direction:

- With **both** settings `true`, editor and terminal both show ligatures.
- With **both** settings `false`, the editor correctly renders raw `=>`, `->`, `!=`, `===` — but the terminal still renders `⇒`, `→`, `≠`, `≡≡≡`.

Same font (`Zed Mono`), same feature names, same file. `buffer_font_features` is honoured live; `terminal.font_features` is not, in either direction. Re-running `cat` in a fresh terminal tab gives the same result.

That would explain the disagreement earlier in this thread — if the terminal renders ligatures unconditionally and ignores the setting, then both "works for me" and "doesn't work for me" are consistent, depending on whether the reporter wanted them on or off.

So the practical complaint (no ligatures in the terminal) looks resolved, but the literal one in the title that the setting does not control ligatures in the terminal still holds.

One thing I did not test: whether a full Zed restart, as opposed to a live settings reload, causes `terminal.font_features` to be picked up.

### Expected
`terminal.font_features` should control OpenType features in the integrated terminal the same way `buffer_font_features` does in the editor — setting `liga`/`calt` to `false` should disable ligatures there.

### Deviations from a default setup
Only the two settings blocks shown above. Default theme and font; `vim_mode` and `relative_line_numbers` are on but neither affects glyph shaping.

cc @yeskunall @AidanV

https://github.com/user-attachments/assets/b6dfeea4-b0a0-4f11-b51f-97898a239726
